engines/python/setup/djl_python/rolling_batch/rolling_batch.py

```diff
@@ -370,6 +369,8 @@ def set_next_token(self,
         # Special handling for error case
         elif finish_reason == "error":
             details_dict["finish_reason"] = finish_reason
+        if self.output_formatter == _json_output_formatter:
+            details_dict["tgi_compat"] = self.tgi_compat
```

Comment:
Why only set when use json?

Reply:
because only applied in json you need to add []